File: Python/Motorcycle.py
# Import class Vehicle yang menjadi parent class
from Vehicle import Vehicle

class Motorcycle(Vehicle):
    def __init__(self, plat_nomor, merk, tahun_produksi, warna, jenis_motor, kapasitas_tangki):
        # Memanggil konstruktor dari parent class (Vehicle)
        super().__init__(plat_nomor, merk, tahun_produksi, warna)
        
        # Menambahkan atribut khusus untuk class Motorcycle
        self.__jenis_motor = jenis_motor
        self.__kapasitas_tangki = kapasitas_tangki

    # Getter methods untuk mendapatkan nilai atribut
    def get_jenis_motor(self):
        return self.__jenis_motor

    def get_kapasitas_tangki(self):
        return self.__kapasitas_tangki

    # Setter methods untuk mengubah nilai atribut
    def set_jenis_motor(self, jenis_motor):
        self.__jenis_motor = jenis_motor

    def set_kapasitas_tangki(self, kapasitas_tangki):
        self.__kapasitas_tangki = kapasitas_tangki

File: CPP/Motorcycle.cpp
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>
#include "Vehicle.cpp"

using namespace std;

class Motorcycle : public Vehicle
{
    private:
        string jenis_motor;
        int kapasitas_tangki;
    public:
    Motorcycle(){
        
    }

    Motorcycle(string jenis_motor, string plat_nomor, string merk, string tahun_produksi, string warna) : Vehicle(string plat_nomor, string merk, string tahun_produksi, string warna){
        this->jenis_motor = jenis_motor;
        kapasitas_tangki ="";
    }
    Motorcycle(string jenis_motor,int kapasitas_tangki, string plat_nomor, string merk, string tahun_produksi, string warna) : Vehicle(string plat_nomor, string merk, string tahun_produksi, string warna){
        this->jenis_motor = jenis_motor;
        this->kapasitas_tangki = kapasitas_tangki;
    }

    void set_jenis_motor(string jenis_motor){
        this->jenis_motor = jenis_motor;
    }

    void set_kapasitas_tangki(int kapasitas_tangki){
        this->kapasitas_tangki = kapasitas_tangki;
    }

    string get_jenis_motor(){
        return jenis_motor;
    }
    int get_kapasitas_tangki(){
        return kapasitas_tangki;
    }
}